catroid/src/main/java/org/catrobat/catroid/embroidery/EmbroideryList.java

```diff
@@ -23,25 +23,94 @@
 
 package org.catrobat.catroid.embroidery;
 
-import android.graphics.PointF;
+import org.catrobat.catroid.content.Sprite;
 
 import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
 
-public class EmbroideryList extends ArrayList{
+public class EmbroideryList extends ArrayList<StitchPoint> {
```

Comment:
Please do not override from ArrayList. In about 99% of all cases this is a violation of the SOLID principles
In this case it violates:
* *Single responsibility principle*:
  * The list creates elements
  * The list has way more logic in, than a simple list should have
* *Liskov subsitution principle*:
  * `add` has side effects that are not to be expected from an arraylist
  * `addAll` does not work
  * `set` does not work
  * `add(index, element)` does not work
  * Most other `ArrayList` functions are broken as well
* *Interface segregation principle*:
  * You only ever use a very small subset of what an `ArrayList` or even a `List` offers, most of it is even unusable

Apart from that, you should prefer composition over inheritance.

I'd like to support you in finding a better solution.
From my point of view you should start at finding a better name for this. It's probably either some kind of Controller or Adapter. Best by starting to create an interface for this class.